odoo/tools/__init__.py

```diff
@@ -10,28 +10,30 @@
     from . import _monkeypatches_urls
 
 from . import appdirs
+from . import arabic_reshaper
 from . import cloc
 from . import constants
 from . import pdf
 from . import pycompat
+from . import template_inheritance
 from . import win32
-from .barcode import *
+from .parse_version import parse_version
```

Comment:
Define `__all__` in all modules and just import * every time. It will be simpler to look for changes in the future.

parse_version is missing the *all* attribute.

Reply:
Since you wrote the `__all__` attributes on sub-modules, it makes it cleaner and more future-proof to import *. That's the main advantage to have that attribute on packages (aside from documentation).
@rco-odoo What do you think?

Reply:
I see your point. You could have a name conflict in the init and making it explicit, you can see it much easier.
Therefore, can you replace the * by explicit imports?

Reply:
The purpose of `__all__` is indeed to make statements with `import *` slightly safer and cleaner.  But I agree that they suck anyway.

Reply:
Ok. Good for me then.